### DISCUSSION TITLE
MacVim r180

### DISCUSSION THREAD

**Discussion by ychin** [macvim-dev/macvim#1500] [Releases]
Updated to Vim 9.1.0727

[![MacVim release-180 download](https://img.shields.io/github/downloads/macvim-dev/macvim/release-180/MacVim.dmg?label=macOS%2010.13%2B)](https://github.com/macvim-dev/macvim/releases/download/release-180/MacVim.dmg) [![MacVim release-180 download (10.9-10.12)](https://img.shields.io/github/downloads/macvim-dev/macvim/release-180/MacVim_10.9.dmg?label=macOS%2010.9-10.12)](https://github.com/macvim-dev/macvim/releases/download/release-180/MacVim_10.9.dmg)

This update mostly syncs to new upstream Vim version, along with small fixes.

Features
====================

New Vim features
--------------------

- Vim now supports the XDG Base Directory Specification. You can now put your vimrc and plugins in `~/.config/vim` intsead of `~/.vim` / `~/.vimrc`. See `:h xdg-base-dir`. [v9.1.0327](https://github.com/vim/vim/commit/c9df1fb35a1866901c32df37dd39c8b39dbdb64a)
- Auto completion
    - Matched text in a completion popup menu will now be highlighted [v9.1.0476](https://github.com/vim/vim/commit/40c1c3317d92f8c2adadf744fab72e4458e2a9fa)
      <img width="147" alt="image" src="https://github.com/user-attachments/assets/9dff8254-74ca-455d-b65e-50ed87e30961">
    - Insert mode fuzzy completion (`:set completopt+=fuzzy`). [v9.1.0463](https://github.com/vim/vim/commit/a218cc6cdabae1113647b817c4eefc2b60a6902f) / [v9.1.0598](https://github.com/vim/vim/commit/8159fb18a92e9a9f5e35201bd92bf651f4d5835c)
- New options:
    - `tabclose`: specifies tab closing behavior. [v9.1.0572](https://github.com/vim/vim/commit/5247b0b92e191a046b034171a3b34031e317735f)
    - `winfixbuf`: fixes a window to a buffer and prevent navigating away from it. [v9.1.0147](https://github.com/vim/vim/commit/215703563757a4464907ead6fb9edaeb7f430bea)
- New functions:
    - `getregion()` / `getregionpos()`: Helps retrieve the text selected in a visual selection. [v9.1.0120](https://github.com/vim/vim/commit/3f905ab3c4f66562f4a224bf00f49d98a0b0da91) [v9.1.0142](https://github.com/vim/vim/commit/19b718828d8d5fab52d94c6cdba694641879ab38) [v9.1.0394](https://github.com/vim/vim/commit/b4757e627e6c83d1c8e5535d4887a82d6a5efdd0)
    - `diff()`: Diff provided strings. [v9.1.0071](https://github.com/vim/vim/commit/fa37835b8c0ed0f83952978fca4c332335ca7c46)
    - `gettext()` / `bindtextdomain()`: Vim plugins can now add localized messages. [v9.1.0509](https://github.com/vim/vim/commit/ce0ef910df837b9b961f007a0a35064cad85188b).
    - `filecopy()`: Copy a file using Vim script. [v9.1.0465](https://github.com/vim/vim/commit/60c8743ab6c90e402e6ed49d27455ef7e5698abe)
- New bundled optional plugins (use `packadd` to enable them):
    - comment: toggle code comments on/off. vim/vim#14634
    - nohlsearch: simple package to automatically set `nohlsearch`. vim/vim#15042
- Highlighting for diff files are back to normal. [v9.1.0016](https://github.com/vim/vim/commit/124371c5a149a8c0c75c04b6c90ac11e71a0aa97)
- Vim9 script improvements.
- Miscellaneous smoothscroll, termdebug, security fixes.
- Other platforms (these don't affect MacVim and included for reference only):
    - GTK Vim GUI can now also use <D-...> to map super keys, similar to MacVim for its Cmd key mappings. This should allow sharing cross-platform configs easier. [v9.1.0058](https://github.com/vim/vim/commit/92e90a1e102825aa9149262cacfc991264db05df)
    - Vim GUI now supports Wayland in Linux. [v9.1.0064](https://github.com/vim/vim/commit/6e0a18f82bd04a45e9682523c42af9cbbc37684c)

Misc
--------------------

- When resizing MacVim, the title bar's message looks a little nicer now. #1488 by @sfsam
- Copyright disclaimer in About MacVim no longer shows a year. #1497

General
====================

- Sparkle (updater for MacVim) is now updated to 2.6.4. For legacy builds it's now updated to 1.27.3.  #1494

Fixes
====================

- Fixed dragging tabs to reorder them resulting in a crash. #1499

<details><summary>Compatibility</summary>

Compatibility
====================

Requires macOS 10.9 or above. (10.9 - 10.12 requires downloading a separate legacy build)

Script interfaces have compatibility with these versions:

- Lua 5.4
- Perl 5.30
- Python2 2.7
- Python3 3.9 or above
- Ruby 3.3

</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/macvim-dev/macvim/releases/tag/release-180'>MacVim r180</a>.</em>

**Comment by austinbkim** [comment]
’preciate the `smoothscroll` fixes.  Out of curiosity, why does MacVim use `<<<` to indicate cut-off lines at the top of the editing window with `smoothscroll` but `@@@` to indicate cut-off lines at the bottom of the editing window with `display=lastline`; why not use `@@@` for both, or `<<<` for the former and `>>>` for the latter, for consistency

**Reply by ychin** [reply]
This is more an upstream Vim thing than a MacVim thing. Vim has always used `@@@` for the cut-off lines, and `smoothscroll` is a much more recent addition. Probably too late to change the default now. 

You can always use [`fillchars`](https://macvim.org/docs/options.txt.html#%27fillchars%27) (it's the "lastline" item) to change that to whatever you like.

**Reply by austinbkim** [reply]
Is the choice of `<<<` for the cut-off lines at the top of the editing window with `smoothscroll` also from upstream Vim; if not, why not use `@@@` for consistency/symmetry with the default `fillchars` for cut-off lines at the bottom of the editing window

**Reply by ychin** [reply]
As I mentioned, yes this is from upstream Vim. Most of MacVim's features are primarily GUI and macOS-integration focused. Unless necessary we aren't going to ship features or change defaults that deviate from vanilla Vim itself. You can also see this if you just use `/usr/bin/vim` (macOS system Vim). `smoothscroll` is not a MacVim specific feature.

I also think `<<<` it a lot more intuitive than `@@@` FWIW. There's a bit of asymmetry here but you can just set the "lastline" `fillchars` to change it to `>` if you want.
